packages/migrate/src/MigrateEngine.ts

```diff
@@ -170,9 +178,16 @@ export class MigrateEngine {
     this.latestSchema = schema
 
     try {
-      const introspectResult = await this.runCommand(
+      const introspectResult: EngineArgs.IntrospectResult = await this.runCommand(
         this.getRPCPayload('introspect', { schema, force, compositeTypeDepth, schemas }),
       )
+      const { views } = introspectResult
+
+      if (views) {
+        const schemaPath = this.schemaPath ?? path.join(process.cwd(), 'prisma')
+        await handleViewsIO({ views, schemaPath })
+      }
+
       return introspectResult
```

Comment:
This is the entrypoint of the I/O fs functionality.

- If `views` is undefined → the `views` preview feature is disabled, and we don't get any view definition from the engines.
- If `views` is defined, but empty → we silently delete everything within the `views` folder
- If `views` is defined -> we write files corresponding to the given views definitions, and delete the rest within the `views` folder

Reply:
I think `views` is always defined now, but is `null` when the preview feature is disabled or there are not views found.
The logic here you added should still work with that as expected.

Reply:
We change this to be what @jkomyno described in this PR:

https://github.com/prisma/prisma-engines/pull/3814

Reply:
I don't think so @pimeys. Didn't we change it to:
- No preview feature = `null`
- Preview feature but no views = `[]`
- Preview feature and views found = `[...]`

Reply:
oh yeah, I don't think this is ever undefined...

Reply:
Confirmed, `views` is never undefined, but can be `null`. The logic doesn't change, but I'll update the type signatures to make it more explicit